app/src/main/kotlin/com/wire/android/ui/home/conversations/messages/QuotedMessage.kt

```diff
@@ -0,0 +1,225 @@
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import com.wire.android.R
+import com.wire.android.model.ImageAsset
+import com.wire.android.ui.common.StatusBox
+import com.wire.android.ui.common.colorsScheme
+import com.wire.android.ui.common.dimensions
+import com.wire.android.ui.common.typography
+import com.wire.android.ui.home.conversations.model.QuotedMessageUIData
+import com.wire.android.ui.theme.wireColorScheme
+import com.wire.android.util.ui.UIText
+
+@Composable
+internal fun QuotedMessage(
+    messageData: QuotedMessageUIData
+) {
+    // Draw content
+    when (val quotedContent = messageData.quotedContent) {
+        QuotedMessageUIData.Invalid -> QuotedInvalid()
+
+        is QuotedMessageUIData.GenericAsset -> QuotedGenericAsset(
+            senderName = messageData.senderName,
+            originalDateTimeText = messageData.originalMessageDateDescription,
+            assetName = quotedContent.assetName
+        )
+
+        is QuotedMessageUIData.DisplayableImage -> QuotedImage(
+            senderName = messageData.senderName,
+            asset = quotedContent.displayable,
+            originalDateTimeText = messageData.originalMessageDateDescription
+        )
+
+        QuotedMessageUIData.Deleted -> QuotedDeleted(
+            senderName = messageData.senderName,
+            originalDateDescription = messageData.originalMessageDateDescription
+        )
+
+        is QuotedMessageUIData.Text -> QuotedText(
+            text = quotedContent.value,
+            editedTimeDescription = messageData.editedTimeDescription,
+            originalDateTimeDescription = messageData.originalMessageDateDescription,
+            senderName = messageData.senderName
+        )
+    }
+}
+
+@Composable
+@Suppress("LongParameterList")
+private fun QuotedMessageContent(
+    senderName: String?,
+    modifier: Modifier = Modifier,
+    endContent: @Composable (modifier: Modifier) -> Unit = {},
+    startContent: @Composable () -> Unit = {},
+    footerContent: @Composable () -> Unit = {},
+    centerContent: @Composable () -> Unit = {}
+) {
```

Comment:
This generic function is responsible to create the outlined box that displays all the Quoted messages.

By default, it displays the sender name (if available).


It provides a way to draw:
- A start content. Not used here, but could be to add a `X` button on the Message composer, when sending reply messages; cc. @trOnk12;
- A center content.
- A footer content (usually the "Original message sent at HH:mm") 
- An end content (the icon/picture on the right of the item).